### PR TITLE
4.0.1 Release notes

### PR DESCRIPTION
**Pull request by JRahnama**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,18 @@
 All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
+## [Hotfix & Stable release 4.0.1] - 2022-01-14
```
[the remark this answers is not in the document]
Comment (JRahnama) [reply CHANGELOG.md:6]: This is a hotfix stable release. why are we taking the Hotfix out? I checked with previous hotfix release and they had the very same title

File: release-notes/4.0/4.0.1.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+- AppContext switch `SuppressInsecureTLSWarning` is added to make users to be able to supress TLS security warning while `Encrypt=false` on connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457) [Read more](#suppress-tls-security-warnings)
+
+### Bug Fixes
+- Fixed the issue with Kerberos authentication using NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
+- Fixed the issue with `SqlLocalDB` instance pipe name. [#1433](https://github.com/dotnet/SqlClient/pull/1433)
+- Fixed the issue with enclave providers not being thread safe. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
+- Updated obsolete API calls in NET6. [#1401](https://github.com/dotnet/SqlClient/pull/1401)
+
+### Suppress TLS security warnings
+When connecting to a SQL server, if a protocol lower than TLS1.2 is negotiated, a security warning is printed to the output. This warning can be supressed by enabling the following `AppContext` switch on the application startup while `Encrypt` is set to `false` on connection string.
```
[the remark this answers is not in the document]
Comment (JRahnama) [reply release-notes/4.0/4.0.1.md:20]: why server is capital? isn't it against MS documentation rule?

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (DavoudEshtehari) [reply CHANGELOG.md:6]: They're old fashion naming. I don't believe we've used it recently! Could you link me to the file you checked?

File: release-notes/4.0/4.0.1.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (DavoudEshtehari) [reply release-notes/4.0/4.0.1.md:20]: As I know, we try to keep it capital in docs.
[Microsoft ADO.NET for SQL Server](https://docs.microsoft.com/en-us/sql/connect/ado-net/microsoft-ado-net-sql-server?view=sql-server-ver15)

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JRahnama) [reply CHANGELOG.md:6]: If you search change logs or release notes for hot fixes you can see that.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,18 @@
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
+## [Stable release 4.0.1] - 2022-01-14
+
+### Added
+
+AppContext switch `SuppressInsecureTLSWarning` is added to make users to be able to supress TLS security warning while `Encrypt=false` on connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457) 
+
+### Fixed
+
+- Fixed the issue with Kerberos authentication using NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
```
Comment (Kaur-Parminder) [inline CHANGELOG.md:14]:
```suggestion
- Fixed the issue with Kerberos authentication using .NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,18 @@
...
+- Fixed the issue with `SqlLocalDB` instance pipe name. [#1433](https://github.com/dotnet/SqlClient/pull/1433)
+- Fixed the issue with enclave providers not being thread safe. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
+- Updated obsolete API calls targeting NET6. [#1401](https://github.com/dotnet/SqlClient/pull/1401)
```
Comment (Kaur-Parminder) [inline CHANGELOG.md:17]:
```suggestion
- Updated obsolete API calls targeting .NET 6. [#1401](https://github.com/dotnet/SqlClient/pull/1401)
```

File: release-notes/4.0/4.0.1.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+## Microsoft.Data.SqlClient 4.0.1 released 14 January 2022
+
+This update brings the below changes over the previous preview release:
+
+### Added
+
+- AppContext switch `SuppressInsecureTLSWarning` is added to make users to be able to supress TLS security warning while `Encrypt=false` on connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457) [Read more](#suppress-tls-security-warnings)
+
+### Bug Fixes
+- Fixed the issue with Kerberos authentication using NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
```
Comment (Kaur-Parminder) [inline release-notes/4.0/4.0.1.md:12]:
```suggestion
- Fixed the issue with Kerberos authentication using .NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,18 @@
 All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
+## [Stable release 4.0.1] - 2022-01-14
+
+### Added
+
+AppContext switch `SuppressInsecureTLSWarning` is added to make users to be able to supress TLS security warning while `Encrypt=false` on connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457)
```
Comment (saul) [inline CHANGELOG.md:10]:
```suggestion
AppContext switch `SuppressInsecureTLSWarning` is added to make users to be able to suppress TLS security warning while `Encrypt=false` on connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457) 
```

Comment (johnnypham) [reply CHANGELOG.md:10]: "
```suggestion
Added AppContext switch `SuppressInsecureTLSWarning` to allow suppression of TLS security warning when using `Encrypt=false` in the connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457) 
```

File: CHANGELOG.md (hunk 3 above)
Comment (johnnypham) [reply CHANGELOG.md:14]:
```suggestion
- Fixed Kerberos authentication failure when using .NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,18 @@
...
+## [Stable release 4.0.1] - 2022-01-14
+
+### Added
+
+AppContext switch `SuppressInsecureTLSWarning` is added to make users to be able to supress TLS security warning while `Encrypt=false` on connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457) 
+
+### Fixed
+
+- Fixed the issue with Kerberos authentication using NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
+- Fixed the issue with `SqlLocalDB` instance pipe name. [#1433](https://github.com/dotnet/SqlClient/pull/1433)
```
Comment (johnnypham) [inline CHANGELOG.md:15]:
```suggestion
- Fixed connection failure when using `SqlLocalDB` instance pipe name. [#1433](https://github.com/dotnet/SqlClient/pull/1433)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,18 @@
...
+- Fixed the issue with enclave providers not being thread safe. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
```
Comment (johnnypham) [inline CHANGELOG.md:16]:
```suggestion
- Fixed a failure when executing concurrent queries requiring enclaves. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
```

**Review by Kaur-Parminder — APPROVED** [review]

File: release-notes/4.0/4.0.1.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+### Bug Fixes
+- Fixed the issue with Kerberos authentication using .NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
+- Fixed the issue with `SqlLocalDB` instance pipe name. [#1433](https://github.com/dotnet/SqlClient/pull/1433)
+- Fixed the issue with enclave providers not being thread safe. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
+- Updated obsolete API calls in NET6. [#1401](https://github.com/dotnet/SqlClient/pull/1401)
+
+### Suppress TLS security warnings
+When connecting to a SQL Server, if a protocol lower than TLS 1.2 is negotiated, a security warning is printed out to the console. This warning can be suppressed by enabling the following `AppContext` switch on the application startup while `Encrypt` is set to `false` on connection string.
+`Switch.Microsoft.Data.SqlClient.SuppressInsecureTLSWarning`
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.1.md:19]:
```suggestion
When connecting to a SQL Server, if a protocol lower than TLS 1.2 is negotiated, a security warning is printed out to the console. This warning can be suppressed by enabling the following `AppContext` switch on the application startup while `Encrypt` is set to `false` on connection string.

`Switch.Microsoft.Data.SqlClient.SuppressInsecureTLSWarning`
```

File: release-notes/4.0/4.0.1.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+## Microsoft.Data.SqlClient 4.0.1 released 14 January 2022
+
+This update brings the below changes over the previous preview release:
+
+### Added
+
+- AppContext switch `SuppressInsecureTLSWarning` is added to make users to be able to supress TLS security warning while `Encrypt=false` on connection string. [#1457](https://github.com/dotnet/SqlClient/pull/1457) [Read more](#suppress-tls-security-warnings)
+
+### Bug Fixes
+- Fixed the issue with Kerberos authentication using .NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.1.md:12]:
```suggestion
### Bug Fixes

- Fixed the issue with Kerberos authentication using .NET 6. [#1411](https://github.com/dotnet/SqlClient/pull/1411)
```

File: release-notes/4.0/4.0.1.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+- Fixed the issue with `SqlLocalDB` instance pipe name. [#1433](https://github.com/dotnet/SqlClient/pull/1433)
+- Fixed the issue with enclave providers not being thread safe. [#1451](https://github.com/dotnet/SqlClient/pull/1451)
+- Updated obsolete API calls in NET6. [#1401](https://github.com/dotnet/SqlClient/pull/1401)
+
+### Suppress TLS security warnings
+When connecting to a SQL Server, if a protocol lower than TLS 1.2 is negotiated, a security warning is printed out to the console. This warning can be suppressed by enabling the following `AppContext` switch on the application startup while `Encrypt` is set to `false` on connection string.
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.1.md:18]:
```suggestion
### Suppress TLS security warnings

When connecting to a SQL Server, if a protocol lower than TLS 1.2 is negotiated, a security warning is printed out to the console. This warning can be suppressed by enabling the following `AppContext` switch on the application startup while `Encrypt` is set to `false` on connection string.
```

**Review by DavoudEshtehari — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (johnnypham) [reply CHANGELOG.md]: Please review my suggestion.

File: CHANGELOG.md (hunk 6 above)
Comment (johnnypham) [reply CHANGELOG.md]: Please review my suggestion.

**Review by johnnypham — APPROVED** [review]

**Pull request merged**